### PR TITLE
task: add initial API integration tests

### PR DESCRIPTION
**Pull request by davidhewitt**
This PR adds some initial integration tests which invoke the CLI as a subprocess and check its output is as expected.

The `REINFER_CLI_TEST_ENDPOINT` and `REINFER_CLI_TEST_TOKEN` secrets will need to be added to the repository settings before CI can succeed.

### CODE REVIEW

**Review by mcobzarenco — APPROVED** [review]
Ah, sorry - I've reviewed both commits here https://github.com/reinfer/cli/pull/23#discussion_r477446634 -- i only had one comment re: not using `println!` to avoid a panic

**Comment by davidhewitt** [comment]
As I've tweaked this just a bit to get CI passing, I'd like one final review before merging.

**Review by tommilligan — APPROVED** [review]
Nothing major, happy as is. Answered my own question about not running integration tests - for posterity, good information as usual is [in the docs](https://doc.rust-lang.org/cargo/commands/cargo-test.html#cargo_test_target_selection) and the invocation I would use is `cargo test --bins --lib` to only run unit tests.

File: cli/tests/test_comments.rs
Comment (davidhewitt) [reply cli/tests/test_comments.rs:15]: Yep I think it would work too, though I personally prefer to have the pair on one line rather than 

```
"--source",
source.indentifier()
```

(I used to logically group flags and their values on one line, but automatic formatting has taken that joy away from me!)

File: cli/Cargo.toml
Comment (davidhewitt) [reply cli/Cargo.toml:18]: Yep as per your answer :)

File: cli/tests/common.rs
Comment (davidhewitt) [reply cli/tests/common.rs:24]: Just me being sloppy... I'll change to an `and_then()` chain.

File: cli/tests/test_comments.rs
Comment (tommilligan) [reply cli/tests/test_comments.rs:15]: Ah, fair enough

**Pull request merged**
